### PR TITLE
Auto bind Store action handlers to action types

### PR DESCRIPTION
**Pull request by levithomason**
Yes, `sotre`s are a thing... :P

### CODE REVIEW

File: dist/TapFlux.js [hunk 1]
```diff
@@ -140,6 +140,7 @@ return /******/ (function(modules) { // webpackBootstrap
 	var _ = __webpack_require__(4);
```
Comment (levithomason) [inline dist/TapFlux.js]: Ignore the dist, of course.

File: examples/tap-flux-chat/gulpfile.js [hunk 2]
```diff
@@ -66,6 +67,7 @@ gulp.task('serve', function(cb) {
 
 gulp.task('watch', function() {
   return gulp.watch([
+    paths.tapFluxDist + '*.js',
     paths.root + 'modules/**/*.js',
```
Comment (levithomason) [inline examples/tap-flux-chat/gulpfile.js]: Reload chat example when the dist is rebuilt.

File: examples/tap-flux-chat/modules/chat/stores/MessageStore.js [hunk 3]
```diff
@@ -6,14 +6,7 @@ var ThreadStore = ChatModule.stores.threadStore;
...
 ChatModule.createStore('messageStore', {
-  initialize: function() {
-    this._messages = {};
-    this.bindActions(
-      constants.ActionTypes.CLICK_THREAD, this.onClickThread,
-      constants.ActionTypes.CREATE_MESSAGE, this.onCreateMessage,
-      constants.ActionTypes.RECEIVE_RAW_MESSAGES, this.onReceiveRawMessages
-    );
-  },
+  _messages: {},
```
Comment (levithomason) [inline examples/tap-flux-chat/modules/chat/stores/MessageStore.js]: Sooo much cleaner! :)  And, we still have the `initialize()` func, if you want it.

File: examples/tap-flux-chat/modules/chat/stores/ThreadStore.js [hunk 4]
```diff
@@ -70,15 +63,14 @@ ChatModule.createStore('threadStore', {
...
 
   onClickThread: function(action) {
-    console.log('ThreadStore onClickThread');
     this._currentID = action.threadID;
     this._threads[this._currentID].lastMessage.isRead = true;
   },
 
-  onReceiveRawMessage: function(action) {
+  onReceiveRawMessages: function(action) {
     this.init(action.rawMessages);
```
Comment (levithomason) [inline examples/tap-flux-chat/modules/chat/stores/ThreadStore.js]: This little typo cost me some time.  After solving made me really really want to write validation for action handlers.  I think we should make `actionTypes` an official thing so we can validate against them.  Added an icebox story for this.

File: src/Store.js [hunk 5]
```diff
@@ -8,55 +9,48 @@ var dispatcher = require('./Dispatcher');
...
 
-  // TODO: split this behemoth up and test it
-  bindActions: function() {
-    var _actions = {};
-    var args = Array.prototype.slice.call(arguments);
-    var self = this;
+  // Register with dispatcher
+  // Handle actions that we have handlers for
+  self._dispatchToken = dispatcher.register(function(action) {
+    var handlerName = 'on' + changeCase.pascalCase(action.type);
```
Comment (levithomason) [inline src/Store.js]: This transform should probably be split out.  So we can test it and use it elsewhere, like validation.  Leaving for future PR.

File: test/Store_test.js [hunk 6]
```diff
@@ -14,32 +14,32 @@ describe('Store', function() {
...
-      var errMessage = 'Store(config) config.initialize must be a function, not: undefined';
-
-      function misuse() {
-        new Store({});
-      }
+    it('assigns config properties to the store', function() {
+      var store = new Store({
+        initialize: _.noop,
+        foo: 'bar',
+      });
```
Comment (levithomason) [inline test/Store_test.js]: `initialize()` is no longer required.  Just runs if present.

File: test/Store_test.js [hunk 7]
```diff
@@ -14,32 +14,32 @@ describe('Store', function() {
...
 
-      expect(misuse).to.throw(errMessage);
+      expect(store).to.have.property('foo', 'bar');
     });
 
-    it('assigns config properties to the store', function() {
+    it("binds config methods to the store's context", function() {
```
Comment (levithomason) [inline test/Store_test.js]: Store(config) config properties are added as is, while methods are bound.

**Comment by eanplatter** [comment]
it's working it's working!!! ༼ つ ◕_◕ ༽つ

**Comment by levithomason** [comment]
Mwwwaaahahahhaahaha... ha..

**Comment by levithomason** [comment]
Starting to become a pretty nice lib, can't wait for doc site and validation!

**Pull request merged**